src/components/Auth/SignUpForm.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { Link, Redirect } from 'react-router-dom';
+import { Box, Button, TextField } from '@material-ui/core/';
+import axios from 'axios';
+import { useAuth } from './AuthContext';
+
+const SignUpForm = ({ toggleActiveForm }) => {
+  const [firstName, setFirstName] = useState('');
```

Comment:
Do you think we should use an object state instead of defining individual states here?

```javascript
const [formObject, setFormObject] = useState({
   firstName: nil,
   lastName: nil,
   username: nil,
   email: nil,
});
```

`errorMessage` and `loggedIn` could be separate as they're different sets of data.

Reply:
Hmm, we could! I'm tempted to leave it for now, and consider refactoring it to use `useReducer`, though, instead of the slight re-org here.